File: Python/guessing_game/guessing_game.py
# comment
import random

secret = random.randint(1,100)
guess = 0
guesses = 0

print "Guess the secret number between 1 and 100!"

while guess != secret:
    guesses = guesses + 1
    guess = input("Enter your guess: ")
    if guess > secret:
        print "Your guess is too high.  Try again!"
    elif guess < secret:
        print "Your guess is too low.  Try again!"
    else:
        print "You guessed the secret in", guesses, "guesses!  It was", secret

File: CPP/guessing game/guessing_game.cpp
/**
  Guessing Game
  Copyright 2009 Travis Crowder
  travis.crowder@spechal.com
  Published under the MIT License
*/
#include <iostream>

int main(int argc, char* argv[]){

  /**
    Create our variable to hold the guess made
  */
  int guess = 0;

  /**
    Let's keep track of the number of guesses made.
  */
  int guesses = 0;

  /**
    Set the secret to be a random number between 1 and 100
    Remember to seed the random number generator
  */
  srand(time(0));
  int secret = rand() % 100 + 1;

  /**
    while our guess is not the secret, get a guess
  */
  while(guess != secret){
    std::cout << "Enter your guess, between 1 and 100: ";
    std::cin >> guess;
    guesses++;
    if(guess == secret){
      /** guess was correct */
      std::cout << "You've figured out the secret in " << guesses;
      if(guesses == 1)
        std::cout << " guess";
      else
        std::cout << " guesses";
      std::cout << "!  It was " << secret << "!" << std::endl;
    } else if(guess > secret){
      /** guess was too high */
      std::cout << "Your guess was too high!  Try again!" << std::endl;
    } else {
      /** our guess can only be lower, logically */
      std::cout << "Your guess was too low!  Try again!" << std::endl;
    }
  }

  return 0;
}
